breathe/directives.py

```diff
@@ -206,6 +208,47 @@ def parse_args(self, function_description):
 
             return args
 
+    def create_function_signature(self, node_stack, project_info, filter_, target_handler,
+                                  mask_factory, directive_args):
+        "Standard render process used by subclasses"
+
+        renderer_factory = DoxygenToRstRendererFactory(
+            self.parser_factory,
+            project_info
+            )
+
+        try:
+            object_renderer = renderer_factory.create_renderer(
+                node_stack,
+                self.state,
+                self.state.document,
+                filter_,
+                target_handler,
+                )
+        except ParserError as e:
+            return format_parser_error("doxygenclass", e.error, e.filename, self.state,
+                                       self.lineno, True)
+        except FileIOError as e:
+            return format_parser_error("doxygenclass", e.error, e.filename, self.state, self.lineno)
+
+        context = RenderContext(node_stack, mask_factory, directive_args)
+        node = node_stack[0]
+        with WithContext(object_renderer, context):
+            # this part should be kept in sync with visit_function in sphinxrenderer
+            name = node.get_name()
+            # assume we are only doing this for C++ declarations
+            declaration = ' '.join([
+                object_renderer.create_template_prefix(node),
+                ''.join(n.astext() for n in object_renderer.render(node.get_type())),
+                name,
+                node.get_argsstring()
+            ])
+        parser = cpp.DefinitionParser(declaration,
+                                      location=self.get_source_info(),
+                                      config=self.config)
+        ast = parser.parse_declaration('function', 'function')
```

Comment:
@jakobandersen Shouldn't we check and handle things based upon the current domain? For example if `breathe_domain_by_extension = {"c": "c", "h": "c"}` with some `header.h` I would think we would need to use `c.DefinitionParser` instead of the `cpp` variant. Similar things may apply to other domains (Python and PHP?), but not sure.

What is the rationale for the C++ assumption?

Reply:
Right, technically it should, and it should probably be generalized later (other parts of this directive should as well). However, only C++ supports overloading, so I'm not sure if even the old code has ever been tested with C, Python, or PHP. I know how to deal with a C version, but for Python and  PHP I don't. Note that the old code almost _always_ breaks the result.

Reply:
I suggest to deal with the domain issue once the rest of the directive is fixed, e.g., to deal with #525.

Reply:
That sounds ok, I will merge this for now. Thanks.